clangd/node-clangd
# @clangd/install: Install/update flows for clangd binary releases.

This module performs two basic actions:
 - install latest clangd binary release from github
 - check whether the latest binary release is newer than the installed one

There are lots of interesting cases and failure modes to deal with in
the various user-facing flows. The most complicated and important flow
is the one that runs on startup:
 - check if clangd is installed and configured
 - offer to install it if not
 - check for updates if so

In order to share this logic, and to test it, interactions with the editor
ui and configuration are abstracted into a UI class which will be
implemented in vscode-clangd, coc-clangd, and is faked for the test.
(I have a prototype version of this working for both editors).
